src/Core/Constants.re

```diff
@@ -12,6 +12,10 @@ let defaultFontFamily = "FiraCode-Regular.ttf";
 let fontAwesomeRegularPath = "FontAwesome5FreeRegular.otf";
 let fontAwesomeSolidPath = "FontAwesome5FreeSolid.otf";
 
+let syntaxEagerMaxLines = 500;
+let syntaxEagerMaxLineLength = 1000;
+let syntaxEagerBudget = 0.25; /* 250 milliseconds */
```

Comment:
Do these really need to be global? We should move towards a configuration infrastructure that supports contributing configurations. We're going to need that for extensions anyway.

Reply:
They're used in two places right now (two places we specify defaults - the default config in `ConfigurationValues.re` and `ConfigurationParser` defaults when parsing).

Supporting contributed configurations would be great! I'm on board with that.

Since we don't have it yet, though, I think there is a trade-off - make these not global and duplicate, or have them global in constants. I'd lean towards the latter to ensure they keep in sync. But would be great to revisit when we have a concept of contributed configurations.

Reply:
Yeah, I think it makes sense to have them be global then.

Reply:
I'm working on revamping the theme infrastructure right now btw, and I think much of that can transfer to configuration as well.

Reply:
Cool @glennsl ! Look forward to seeing what you come up with.